Automattic/wp-calypso
# Reader full post - use scss variables to determine breakpoint and fix some layout issues.

Related to # follow up to https://github.com/Automattic/wp-calypso/pull/91990

## Proposed Changes

* Moves the horizontal widths that determine the reader full post page's breakpoint to hide the author details to scss variables. This means that if someone wants to adjust the padding, max-width, etc. in the future the author details should still continue to be hidden when they should and not require the future code contributor to catch an edge case in the breakpoints and manually update a media query.
* Also makes the back button background transparent, on safari on low widths the white background was obstructing the title/subtitle.
* Updates the flex definitions for the content frame to only apply when needed. And to apply margin 0 auto to the story container when we are not at the flex width.  This fixes issues with some of the content overflowing off the page at smaller widths when approaching the limit of where we show the global sidebar as fixed.


BEFORE (Below is Safari - note chrome would do similar but it would bleed off the right instead of the left).
<img width="803" alt="Screenshot 2024-06-27 at 10 17 08 AM" src="https://github.com/Automattic/wp-calypso/assets/28742426/267a142f-10c6-45c0-aef2-6680b67492f3">


AFTER
<img width="782" alt="Screenshot 2024-06-27 at 10 32 29 AM" src="https://github.com/Automattic/wp-calypso/assets/28742426/26ae5b26-4324-4a12-89ad-8f50c0da4482">


## Why are these changes being made?


* Ease of use for future developers. The author details are to be hidden when they can no longer fit side by side with the content/story. But this breakpoint is a calculation of widths and paddings of various items on the screen. To avoid requiring a magic number in the media query to be rewritten every time we update one of these values, we use a variable that adds these values instead. This needs to be a scss variable and not based on css variables since media queries do not support css variables.
* Fix issues with layout on some breakpoint ranges/browsers.

## Testing Instructions



* Run this calypso branch and visit the full post page in the reader.
* Verify the author details continues to disappear once it would no longer fit side-by-side with the story (instead of flex wrapping be above the story).
* There should be no visible change from produciton.

Bonus - adjust values for the initial scss definitions for widths/paddings and verify the above is still true.

* Reduce screen width to just about the point the sidebar would disappear. Verify none of the story content is overflowing off the page.

## Pre-merge Checklist
